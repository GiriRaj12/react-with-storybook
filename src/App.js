import React from 'react';
import './App.css';
import {
  BrowserRouter as Router,
  Switch,
  Route,
} from "react-router-dom"
import Dashboard from './Components/Dashboard.js';
import SidebarContent from './Components/SidebarContent';
import { faChartBar } from '@fortawesome/free-solid-svg-icons';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';

function App() {
  return (
    <Router>
      <div className="App">
        <header className="app-header">
        </header>
        <nav className="side-bar">
          <div className="side-bar-me">
            <SidebarContent renderIcon={<FontAwesomeIcon
              icon={faChartBar} />} content="Dashboard" toRe="/dashboard" />
          </div>
        </nav>
        <Switch>
          <Route path="/dashboard">
            <Dashboard></Dashboard>
          </Route>
          <Route path="/">
            <Dashboard></Dashboard>
          </Route>
        </Switch>
      </div>
    </Router>
  );
}

export default App;
