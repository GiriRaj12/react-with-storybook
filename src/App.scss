.App {
  text-align: center;
}

.side-bar{
  height: 100%;
  background-color: #23263c;
  position: fixed;
  color: #dfe5eb;
  width: 200px;
}

.side-bar-me{
  margin-top: 200px;
}

.sidebar-content-name{
  margin-left: 10px;
}

.sidebar-content{
  text-decoration: none;
  color: white;
  display: flex;
  font-size: 20px;
  font-weight: 500;
  text-align: left;
  padding: 15px;
  margin-top: 10px;
  background-color:#23263c;
  cursor: pointer;
} 

.sidebar-content:hover{
  background-color: #44484b;
}