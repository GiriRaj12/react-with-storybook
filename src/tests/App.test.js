import React from 'react';
import renderer from 'react-test-renderer';
import App from '../App.js';

test('App Snapshot test', () => {
    const render = renderer.create(<App></App>)
    const element = render.toJSON();
    expect(element).toMatchSnapshot();
})