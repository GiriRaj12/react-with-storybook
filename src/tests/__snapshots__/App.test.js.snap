// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Snapshot test 1`] = `
<div
  className="App"
>
  <header
    className="app-header"
  />
  <nav
    className="side-bar"
  >
    <div
      className="side-bar-me"
    >
      <a
        className="sidebar-content"
        href="/dashboard"
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-chart-bar fa-w-16 "
          data-icon="chart-bar"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M332.8 320h38.4c6.4 0 12.8-6.4 12.8-12.8V172.8c0-6.4-6.4-12.8-12.8-12.8h-38.4c-6.4 0-12.8 6.4-12.8 12.8v134.4c0 6.4 6.4 12.8 12.8 12.8zm96 0h38.4c6.4 0 12.8-6.4 12.8-12.8V76.8c0-6.4-6.4-12.8-12.8-12.8h-38.4c-6.4 0-12.8 6.4-12.8 12.8v230.4c0 6.4 6.4 12.8 12.8 12.8zm-288 0h38.4c6.4 0 12.8-6.4 12.8-12.8v-70.4c0-6.4-6.4-12.8-12.8-12.8h-38.4c-6.4 0-12.8 6.4-12.8 12.8v70.4c0 6.4 6.4 12.8 12.8 12.8zm96 0h38.4c6.4 0 12.8-6.4 12.8-12.8V108.8c0-6.4-6.4-12.8-12.8-12.8h-38.4c-6.4 0-12.8 6.4-12.8 12.8v198.4c0 6.4 6.4 12.8 12.8 12.8zM496 384H64V80c0-8.84-7.16-16-16-16H16C7.16 64 0 71.16 0 80v336c0 17.67 14.33 32 32 32h464c8.84 0 16-7.16 16-16v-32c0-8.84-7.16-16-16-16z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
        <span
          className="sidebar-content-name"
        >
          Dashboard
        </span>
      </a>
    </div>
  </nav>
  <div
    className="container-background"
  >
    <div
      className="user-details"
    >
      <div
        className="name-conatiner"
      >
        G
      </div>
      <span
        className="user-name-details"
      >
        Giriraj
      </span>
    </div>
    <div
      className="objective-container"
    >
      Loading...
    </div>
    <div
      className="external-graphs-container"
    >
      Loading...
    </div>
  </div>
</div>
`;
